main.go

```diff
@@ -514,7 +516,8 @@ func typeDisplayName(t *types.Type, c generatorConfig, typePkgMap map[*types.Typ
 	}
 
 	if t.Kind == types.Slice {
-		s = "[]" + s
+		t = tryDereference(t)
+		s = "[]" + t.Name.Name
```

Comment:
do you mind reminding what this part does

Reply:
Yes, if the Type is slice then this code prefixes "[]" to the type name. 

For example - if the type is `[]GCPDisk`, then `tryDeference(t)` would return `GCPDisk` and this code prefixes "[]". Thus, rendering the name on output file as `[]GCPDisk`

Reply:
Hm but wasn't it already doing that?

Reply:
@ahmetb Yes. I now realise no changes are required here in this part of the code. Sorry about the residue code which was uncleaned.

Reply:
OK I am glad we changed just 2 words and still got this done. 👍🏼

Reply:
Yusssss! Thank you so much for guiding me through. I am very glad to mark this as my FIRST open source contribution outside my org 🥳